test/16_recode.jl

```diff
@@ -534,4 +534,12 @@ end
     @test !isordered(y)
 end
 
+@testset "Missings.replace should work on CategoricalArrays" begin
+    x = categorical(["a", "b", missing, "a"])
+    y = categorical(["a", "b", "", "a"])
+    r = Missings.replace(x, "")
```

Comment:
Would be nice to test replacing with an existing level, and also what happens when the replacement cannot be converted the type element type.